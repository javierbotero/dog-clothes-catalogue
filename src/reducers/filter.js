import { FILTER } from '../constants/constants';

const filter = (state = 'All', action) => {
  if (action.type === FILTER) {
    return action.payload;
  }
  return state;
};

export default filter;
