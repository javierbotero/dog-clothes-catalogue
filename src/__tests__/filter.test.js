import filter from '../reducers/filter';
import * as deps from './test-util/test-util';

describe('Test filter reducer', () => {
  it('Check correct initial state', () => {
    const result = filter(undefined, '');
    expect(result).toBe('All');
  });
  it('Check change of initial state for New', () => {
    const result = filter('All', { type: deps.FILTER, payload: 'New' });
    expect(result).toBe('New');
  });
});
