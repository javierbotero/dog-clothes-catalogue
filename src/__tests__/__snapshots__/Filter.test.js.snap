// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filter renders correctly 1`] = `
<div
  className="page filter"
>
  <main>
    <select
      onChange={[MockFunction]}
      value="All"
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Coats and Jackets"
      >
        Coats and Jackets
      </option>
      <option
        value="Pow Protection"
      >
        Pow Protection
      </option>
      <option
        value="Sweaters and Hoodies"
      >
        Sweaters and Hoodies
      </option>
      <option
        value="Cold Weather Gear"
      >
        Cold Weather Gear
      </option>
      <option
        value="Jerseys"
      >
        Jerseys
      </option>
      <option
        value="Pajamas"
      >
        Pajamas
      </option>
    </select>
    <div
      className="products"
    >
      <div
        className="container-product"
      >
        <h3>
          product1
        </h3>
        Categeory:
        <span>
          Coats and Jackets
        </span>
        <br />
        Price:
        <span>
          100
        </span>
        <br />
        <div
          className="pictures-filter"
          style={
            Object {
              "backgroundImage": "url(coat1)",
            }
          }
        />
        <a
          className="more-info"
          href="/2"
        >
          More Info
        </a>
      </div>
    </div>
  </main>
  <aside>
    <div>
      <div>
        Get an special gift for purchases over $200
      </div>
      <div>
        <div
          className="picture-banner"
          style={
            Object {
              "backgroundImage": "url(coat1)",
            }
          }
        />
        <span>
          100
        </span>
        <br />
        <span>
          product1
        </span>
        <br />
      </div>
      <div>
        <div
          className="picture-banner"
          style={
            Object {
              "backgroundImage": "url(coat1)",
            }
          }
        />
        <span>
          100
        </span>
        <br />
        <span>
          product1
        </span>
        <br />
      </div>
      <a
        className="more-info"
        href="/"
      >
        Special Gift
      </a>
    </div>
  </aside>
</div>
`;
